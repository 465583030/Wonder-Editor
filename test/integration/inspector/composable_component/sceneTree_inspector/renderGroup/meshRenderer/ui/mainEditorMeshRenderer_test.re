open Wonder_jest;

open Expect;

open Expect.Operators;

open AssetNodeType;

open Sinon;

let _ =
  describe("mainEditor meshRenderer", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());
      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => {
      restoreSandbox(refJsObjToSandbox(sandbox^));
      StateEditorService.getState()
      |> AssetCurrentNodeDataEditorService.clearCurrentNodeData
      |> AssetCurrentNodeParentIdEditorService.clearCurrentNodeParentId
      |> StateEditorService.setState
      |> ignore;
    });

    /* TODO remove "prepare currentSelectSource"
    (test "test meshRenderer change drawMode" directly) */
    describe("prepare currentSelectSource", () => {
      beforeEach(() => {
        MainEditorSceneTool.createDefaultScene(
          sandbox,
          MainEditorSceneTool.setFirstBoxTobeCurrentSceneTreeNode,
        );

        CurrentSelectSourceEditorService.setCurrentSelectSource(
          EditorType.SceneTree,
        )
        |> StateLogicService.getAndSetEditorState;
      });

      describe("test meshRenderer change drawMode", () => {
        describe("test set drawMode to Lines", () => {
          test("test snapshot", () => {
            let lineType = MainEditorMeshRendererTool.getDrawModeLineType();

            MainEditorMeshRendererTool.triggerChangeDrawModeEvent(lineType);

            BuildComponentTool.buildMeshRenderer(
              TestTool.buildEmptyAppState(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });

          test("test logic", () => {
            let lineType = MainEditorMeshRendererTool.getDrawModeLineType();

            MainEditorMeshRendererTool.triggerChangeDrawModeEvent(lineType);

            let meshRenderer =
              GameObjectTool.getCurrentGameObjectMeshRenderer();

            MeshRendererEngineService.getDrawMode(meshRenderer)
            |> StateLogicService.getEngineStateToGetData
            |> expect == lineType;
          });
        });
        describe("test set drawMode to Points", () => {
          test("test snapshot", () => {
            let pointType = MainEditorMeshRendererTool.getDrawModePointType();

            MainEditorMeshRendererTool.triggerChangeDrawModeEvent(pointType);

            BuildComponentTool.buildMeshRenderer(
              TestTool.buildEmptyAppState(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });

          test("test logic", () => {
            let pointType = MainEditorMeshRendererTool.getDrawModePointType();

            MainEditorMeshRendererTool.triggerChangeDrawModeEvent(pointType);

            let meshRenderer =
              GameObjectTool.getCurrentGameObjectMeshRenderer();

            MeshRendererEngineService.getDrawMode(meshRenderer)
            |> StateLogicService.getEngineStateToGetData
            |> expect == pointType;
          });
        });
        describe("test set drawMode to Triangle_fan", () => {
          test("test snapshot", () => {
            let triangleFanType =
              MainEditorMeshRendererTool.getDrawModeTriangleFanType();

            MainEditorMeshRendererTool.triggerChangeDrawModeEvent(
              triangleFanType,
            );

            BuildComponentTool.buildMeshRenderer(
              TestTool.buildEmptyAppState(),
            )
            |> ReactTestTool.createSnapshotAndMatch;
          });

          test("test logic", () => {
            let triangleFanType =
              MainEditorMeshRendererTool.getDrawModeTriangleFanType();

            MainEditorMeshRendererTool.triggerChangeDrawModeEvent(
              triangleFanType,
            );

            let meshRenderer =
              GameObjectTool.getCurrentGameObjectMeshRenderer();

            MeshRendererEngineService.getDrawMode(meshRenderer)
            |> StateLogicService.getEngineStateToGetData
            |> expect == triangleFanType;
          });
        });
      });
    });
  });