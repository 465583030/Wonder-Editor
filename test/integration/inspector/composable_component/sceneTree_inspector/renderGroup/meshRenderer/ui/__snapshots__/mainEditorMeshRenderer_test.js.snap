// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Line test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`MainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Lines test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`MainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Points test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`MainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Points test snapshot 2`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="6"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`MainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set wrapT to MIRRORED_REPEAT test snapshot 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <article
      className="wonder-texture-assetTree"
    >
      <div
        className=""
      >
        <h1>
          Texture
        </h1>
        <hr />
        <div
          className=""
        >
          <article
            className="wonder-string-input"
          >
            <span
              className="component-label"
            >
              name : 
            </span>
            <input
              className="input-component float-input"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
              value="texture3"
            />
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapS Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              WrapT Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="2"
              >
                REPEAT
              </option>
              <option
                value="1"
              >
                MIRRORED_REPEAT
              </option>
              <option
                value="0"
              >
                CLAMP_TO_EDGE
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Mag Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="1"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
        <div
          className=""
        >
          <article
            className="wonder-select"
          >
            <span
              className=""
            >
              Min Filter Mode
            </span>
            <select
              onChange={[Function]}
              value="0"
            >
              <option
                value="0"
              >
                NEAREST
              </option>
              <option
                value="1"
              >
                LINEAR
              </option>
              <option
                value="2"
              >
                NEAREST_MIPMAP_NEAREST
              </option>
              <option
                value="3"
              >
                LINEAR_MIPMAP_NEAREST
              </option>
              <option
                value="4"
              >
                NEAREST_MIPMAP_LINEAR
              </option>
              <option
                value="5"
              >
                LINEAR_MIPMAP_LINEAR
              </option>
            </select>
          </article>
        </div>
      </div>
    </article>
  </article>
</article>
`;

exports[`mainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Lines test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`mainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Points test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`mainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Points test snapshot 2`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="6"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;

exports[`mainEditor meshRenderer prepare currentSelectSource test meshRenderer change drawMode test set drawMode to Triangle_fan test snapshot 1`] = `
<article
  className="wonder-mesh-renderer"
>
  <div
    className=""
  >
    <article
      className="wonder-select"
    >
      <span
        className=""
      >
        draw mode
      </span>
      <select
        onChange={[Function]}
        value="6"
      >
        <option
          value="0"
        >
          points
        </option>
        <option
          value="1"
        >
          lines
        </option>
        <option
          value="2"
        >
          line_loop
        </option>
        <option
          value="3"
        >
          Line_strip
        </option>
        <option
          value="4"
        >
          Triangles
        </option>
        <option
          value="5"
        >
          triangle_strip
        </option>
        <option
          value="6"
        >
          Triangle_fan
        </option>
      </select>
    </article>
  </div>
</article>
`;
